src/TestQuaternion.java

```diff
@@ -192,4 +202,95 @@ public void testMultiplicationQuaternion() {
         assertEquals(expected, new Quaternion(8, -4, -2, -6));
 
     }
+
+    /**
+     * Tester for the conjugation method.
+     */
+    public void testConjugation() {
+        assertEquals(new Quaternion(1, -1, -1, -1), new Quaternion(1, 1, 1, 1).conjugate());
+        assertEquals(new Quaternion(1, 1, 1, 1), new Quaternion(1, -1, -1, -1).conjugate());
+        assertEquals(new Quaternion(1, 1, 1, 1), new Quaternion(1, -1, -1, -1).conjugate());
+
+    }
+
+    /**
+     * Tester for the conjugateMultiplying method.
+     */
+    public void testConjugateMultiplying() {
+        assertEquals(new Quaternion(1, -1, -1, -1),
+                new Quaternion(1, 1, 1, 1).conjugateMultiplying());
+        assertEquals(new Quaternion(1, 1, 1, 1),
+                new Quaternion(1, -1, -1, -1).conjugateMultiplying());
+        assertEquals(new Quaternion(1, 1, 1, 1),
+                new Quaternion(1, -1, -1, -1).conjugateMultiplying());
+
+        // Tests for equality between the two forms of conjugation.
+        assertEquals(new Quaternion(1, 2, 3, 4).conjugate(),
+                new Quaternion(1, 2, 3, 4).conjugateMultiplying());
+
+    }
+
+    /**
+     * Tester for the norm method.
+     */
+    public void testNorm() {
+
+        double tol = 1e-6; // Tolerance.
+        assertEquals(new Quaternion(1, 1, 1, 0).norm(), Math.sqrt(3), tol);
+        assertEquals(new Quaternion(1, 0, 0, 0).norm(), 1, tol);
+        assertEquals(new Quaternion(-1, 1, -1, 1).norm(), 2, tol);
+        assertEquals(new Quaternion(1, 2, -3, 4).norm(), Math.sqrt(30), tol);
+        assertEquals(new Quaternion(2, 2, 2, 2).norm(), 4, tol);
+
+        try {
```

Comment:
+1 on testing the exception! Note however that there are some built-in ways of doing that in JUnit http://stackoverflow.com/questions/15216438/junit-testing-exceptions